### PR TITLE
feat: Introduce onAttachmentTap on messages attachment

### PR DESCRIPTION
**Pull request by Abacaxi-Nelson**
# Submit a pull request

## CLA

- [ X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [X ] The code changes follow best practices
- [ X] Code changes are tested (add some information if not applicable)

## Description of the pull request

UseCase : Customize messages attachment, and display them as you want

### CODE REVIEW

**Comment by imtoori** [comment]
Thanks a lot!
I'll review it as soon as possible

also, I changed the base branch to `develop`, that's where we usually work before publishing to pub.dev

I'll write some instruction on the readme so that for the next prs people know our workflow

**Comment by Abacaxi-Nelson** [comment]
@imtoori 
I have another small PR (add custom Icons button for attachment on MessageInput, and enable/disable existing IconButton).
Do you want me to add this to the current PR, or wait for instruction on the readme about develop branch, before submitting it ?

**Comment by imtoori** [comment]
@Abacaxi-Nelson 
let's open another pr for these other changes

about the contributing guide, that will take a bit of time. Just keep in mind to create the branch from `develop` and not from `master`

Let me know if there is anything else I can do to help and thanks again!

**Comment by imtoori** [comment]
@Abacaxi-Nelson can I ask you to fix the conflicts and rename the pr please?

**Comment by imtoori** [comment]
I tried to fix conflicts using the web editor

File: packages/stream_chat_flutter/lib/src/message_list_view.dart [hunk 1]
```diff
@@ -110,36 +110,37 @@ class MessageDetails {
...
+      this.loadingBuilder,
+      this.emptyBuilder,
+      this.systemMessageBuilder,
+      this.messageListBuilder,
+      this.errorWidgetBuilder,
+      this.messageFilter,
+      this.customAttachmentBuilders,
+      this.onMessageTap,
+      this.onSystemMessageTap,
+      this.onAttachmentTap})
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_list_view.dart:142]: missing trailing comma

File: packages/stream_chat_flutter/lib/src/attachment/video_attachment.dart [hunk 2]
```diff
@@ -11,16 +11,18 @@ class VideoAttachment extends AttachmentWidget {
...
-  }) : super(key: key, message: message, attachment: attachment, size: size);
+  const VideoAttachment(
+      {Key key,
+      @required Message message,
+      @required Attachment attachment,
+      Size size,
+      this.messageTheme,
+      this.onShowMessage,
+      this.onReturnAction,
+      this.onAttachmentTap})
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/attachment/video_attachment.dart:24]: can you add a trailing comma?
https://flutter.dev/docs/development/tools/formatting#using-trailing-commas

File: packages/stream_chat_flutter/lib/src/message_widget.dart [hunk 3]
```diff
@@ -146,104 +146,110 @@ class MessageWidget extends StatefulWidget {
...
+      this.readList,
+      this.padding,
+      this.textPadding = const EdgeInsets.symmetric(
+        horizontal: 16.0,
+        vertical: 8.0,
+      ),
+      this.attachmentPadding = EdgeInsets.zero,
+      this.allRead = false,
+      this.onQuotedMessageTap,
+      this.onAttachmentTap})
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:201]: missing trailing comma

File: packages/stream_chat_flutter/lib/src/message_widget.dart [hunk 4]
```diff
@@ -146,104 +146,110 @@ class MessageWidget extends StatefulWidget {
...
+                messageTheme: messageTheme,
+                size: Size(
+                  MediaQuery.of(context).size.width * 0.8,
+                  MediaQuery.of(context).size.height * 0.3,
+                ),
+                onShowMessage: onShowMessage,
+                onReturnAction: onReturnAction,
+                onAttachmentTap: () {
+                  onAttachmentTap(message, attachment);
+                });
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:216]: missing trailing comma

File: packages/stream_chat_flutter/lib/src/message_widget.dart [hunk 5]
```diff
@@ -146,104 +146,110 @@ class MessageWidget extends StatefulWidget {
...
+                size: Size(
+                  MediaQuery.of(context).size.width * 0.8,
+                  MediaQuery.of(context).size.height * 0.3,
+                ),
+                message: message,
+                onShowMessage: onShowMessage,
+                onReturnAction: onReturnAction,
+                onAttachmentTap: () {
+                  onAttachmentTap(message, attachment);
+                });
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:231]: missing trailing comma

File: packages/stream_chat_flutter/lib/src/message_widget.dart [hunk 6]
```diff
@@ -146,104 +146,110 @@ class MessageWidget extends StatefulWidget {
...
-            );
+                attachment: attachment,
+                messageTheme: messageTheme,
+                message: message,
+                size: Size(
+                  MediaQuery.of(context).size.width * 0.8,
+                  MediaQuery.of(context).size.height * 0.3,
+                ),
+                onShowMessage: onShowMessage,
+                onReturnAction: onReturnAction);
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:243]: missing trailing comma
also, how come this has no onAttachmentTap param?

File: packages/stream_chat_flutter/lib/src/message_widget.dart [hunk 7]
```diff
@@ -146,104 +146,110 @@ class MessageWidget extends StatefulWidget {
...
-                MediaQuery.of(context).size.width * 0.8,
-                MediaQuery.of(context).size.height * 0.3,
-              ),
-            );
+                message: message,
+                attachment: attachment,
+                size: Size(
+                  MediaQuery.of(context).size.width * 0.8,
+                  MediaQuery.of(context).size.height * 0.3,
+                ));
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_widget.dart:252]: missing trailing comma
also, how come this has no onAttachmentTap param?

**Review by imtoori — CHANGES_REQUESTED** [review]
Some small things to fix, but overall it's good 
Thanks for this and looking forward to merging it

File: packages/stream_chat_flutter/lib/src/attachment/video_attachment.dart (hunk 2 above)
Comment (Abacaxi-Nelson) [reply packages/stream_chat_flutter/lib/src/attachment/video_attachment.dart]: done

File: packages/stream_chat_flutter/lib/src/message_widget.dart (hunk 3 above)
Comment (Abacaxi-Nelson) [reply packages/stream_chat_flutter/lib/src/message_widget.dart]: done

File: packages/stream_chat_flutter/lib/src/message_widget.dart (hunk 4 above)
Comment (Abacaxi-Nelson) [reply packages/stream_chat_flutter/lib/src/message_widget.dart]: done

File: packages/stream_chat_flutter/lib/src/message_widget.dart (hunk 5 above)
Comment (Abacaxi-Nelson) [reply packages/stream_chat_flutter/lib/src/message_widget.dart]: done

File: packages/stream_chat_flutter/lib/src/message_widget.dart (hunk 6 above)
Comment (Abacaxi-Nelson) [reply packages/stream_chat_flutter/lib/src/message_widget.dart:245]: i find no value to do it :)

File: packages/stream_chat_flutter/lib/src/message_widget.dart (hunk 7 above)
Comment (Abacaxi-Nelson) [reply packages/stream_chat_flutter/lib/src/message_widget.dart:254]: i find no value to do it :)

**Review by imtoori — APPROVED** [review]
LGTM

**Pull request merged**